src/vivarium/framework/results/manager.py

```diff
@@ -26,157 +165,99 @@ class ResultsInterface:
-    we may want to track all the heart attacks that occur for 50-55 year old
-    men in a several year simulation of cardiovascular health. Each time step,
-    the results system will call the producer the user has registered to count
-    heart attacks by age group and that producer will count all the heart
-    attacks in that particular time step. The results system will then add
-    that time step's worth of heart attacks for 50-55 year old men to its
-    running total.
+    The purpose of this interface is to provide controlled access to a results
+    backend by means of the builder object. It exposes methods
+    to register stratifications, set default stratifications, and register
+    results producers. There is a special case for stratifications generated
+    by binning continuous data into categories.
 
+    The expected use pattern would be for a single component to register all
+    stratifications required by the model using :func:`register_default_stratifications`,
+    :func:`register_stratification`, and :func:`register_binned_stratification`
+    as necessary. A “binned stratification” is a stratification special case for
+    the very common situation when a continuous value needs to be binned into
+    categorical bins. The is_vectorized argument should be True if the mapper
+    function expects a DataFrame, and False if it expects a row of the DataFrame
+    and should be used by calling df.apply.
     """
 
     def __init__(self, manager: ResultsManager) -> None:
         self._manager = manager
+        self._name = "results_interface"
 
-    def add_mapping_strategy(
-        self, new_column: str, mapper: Callable[[pd.Index], pd.Series]
-    ) -> None:
-        """Adds a specification to map simulation state into a new column.
+    @property
+    def name(self) -> str:
+        """The name of this ResultsInterface."""
+        return self._name
 
-        This allows a user to specify an arbitrary function to generate
-        new columns in the expanded state table for use in results production.
+    def set_default_stratifications(self, default_stratifications: List[str]):
+        self._manager.set_default_stratifications(default_stratifications)
 
-        For instance, suppose you have a column `time_of_death` that records
-        the time a simulant dies as a time stamp.  In order to stratify
-        results, you might map that column to a `year_of_death` column to
-        group deaths by the year in which they occurred.
+    # TODO: It is not reflected in the sample code here, but the “when” parameter should be added
+    #  to the stratification registration calls, probably as a List.
+    def register_stratification(
+        self,
+        name: str,
+        categories: List[str],
+        mapper: Callable = None,
+        is_vectorized: bool = False,
+        requires_columns: List[str] = (),
+        requires_values: List[str] = (),
+    ) -> None:
+        # TODO: Fill in the XXX below!
```

Comment:
I think these XXX should be filled out in this ticket. We can talk about what these parameters represent if it is unclear.